common-model/src/main/java/com/sequenceiq/common/api/type/ScalingMode.java

```diff
@@ -0,0 +1,20 @@
+package com.sequenceiq.common.api.type;
+
+/**
+ * Specifies whether an instance group supports scaling option.
+ * Instance group with auto scaling mode is both auto scalable and manually scalable.
+ * None means the instance group is not scalable.
+ * If unspecified, the instance group can be scaled manually. This is the current behaviour for backward compatibility.
+ * Unless specified none, the instance groups can be scaled manually.
+ */
+public enum ScalingMode {
+    MANUAL, AUTO, NONE, UNSPECIFIED;
+
+    public boolean isAutoscalable() {
```

Comment:
Manually Scalable hostGroups would also be AutoScalable right, should Manual\Auto be combined into one enum ? Also considering custom templates would there be validation that a hostGroup marked Auto\Manual contains only scalable services ?

Reply:
Hi @smaniraju , Not all manually scalable hostGroups would be auto-scalable. For example gateway nodes. We do not have a mechanism to detect auto-scalability. The converse is true. Hence they are split into two different enums. Regarding your other question, the idea to put this in cluster template instead of hard coding it, is to give flexibility for the cluster shapes to choose auto-scalable components by themselves. For example, an OpDB template may choose some of the HBase node groups to be auto-scalable. I do not know how the backend would work, but this change enables such functionality.

Reply:
Removed manual and auto enums for now. This will unblock UI efforts. Will send PR for auto or algo related scalability enum to the specific branch.

Reply:
why did you have to remove those?

Reply:
So, the differentiation of manual and auto is unclear. Even within the auto, there are rule-based and schedule based differences. To solve DISTX-427, I wanted to get this in quicker so that UI for the Resize button can be adjusted in the master branch.

For the auto/algo/time/rule differences can put that in the auto-scaling branch for auto-scaling related UI efforts DISTX-421.